pydap_extras/handlers/pcic.py

```diff
@@ -230,35 +265,34 @@ class RawPcicSqlHandler(PcicSqlHandler):
     virtual = True
 
     def get_full_query(self, stn_id, sesh):
-        """Sends a special query to the database that actually retrieves generated SQL for constructing an observation table (time by variable) for a single station. The query needs to return at least one column (obs_time) with additional columns for each available variable, if any. Uses the ``query_one_station`` stored procedure.
+        """
+        Retrieves generated SQL for constructing an observation table (time by variable)
+        for a single station. The query needs to return at least one column (obs_time)
+        with additional columns for each available variable, if any. Uses the
+        ``query_one_station`` stored procedure.
+
         :param stn_id: the *database* station_id of the desired station
         :type stn_id: int or str
-        :param sesh: an sqlalchemy session
+        :param sesh: a sqlalchemy session
         """
 
+        # TODO: What is this for? A comment at least would help.
         if not self.get_vars(stn_id, sesh):
             return "SELECT obs_time FROM obs_raw WHERE NULL"
 
         query_string = "SELECT query_one_station(%s)" % stn_id
         return sesh.execute(query_string).fetchone()[0]
 
     def get_vars(self, stn_id, sesh):
-        """Makes a database query to retrieve all of the raw variables for a particular station"""
+        """Retrieves all raw (observation) variables for a particular station"""
         q = (
             sesh.query(Variable)
-            .join(VarsPerHistory)
-            .join(History)
-            .join(Station)
-            .join(Network)
+            .join(VarsPerHistory, VarsPerHistory.vars_id == Variable.id)
+            .join(History, History.id == VarsPerHistory.history_id)
+            .join(Station, Station.id == History.station_id)
+            .join(Network, Network.id == Station.network_id)  # Or join to Variable
             .filter(Station.id == stn_id)
-            .filter(
-                not_(
-                    or_(
-                        Variable.cell_method.like("%within%"),
-                        Variable.cell_method.like("%over%"),
-                    )
-                )
-            )
+            .filter(variable_tags.contains(array(["observation"])))
```

Comment:
The magic! Looks great!

FWIW, I haven't found anywhere obvious (say in `pycds`?) where the usage of `variable_tags` is documented and I wouldn't say that it's overwhelmingly obvious to me.

Reply:
Excellent point. New [issue](https://github.com/pacificclimate/pycds/issues/187).